FusionIIIT/applications/otheracademic/models.py

```diff
@@ -35,11 +35,109 @@ class LeaveFormTable(models.Model):
     leave_type = models.CharField(max_length=20, choices=LEAVE_TYPES)
     approved = models.BooleanField()
     rejected = models.BooleanField()
+    hod= models.CharField(max_length=100)
```

Comment:
This is a design decision that needs to be clarified, documentation regarding what is HOD attribute is missing and if it's head of department, why is this field a char field instead of a Foreign Key?

Reply:
Basically while running the sdk api of create file ,there is problem of fetching user designation as it was not available in database .So, i ask the credential that is user id of hod as charfield and after that i use the below function to get the user_id that is foreign key
 hod_user=User.objects.get(username=leave_entry.hod)
 Basically ,i follow the methods of filetracking module that ask the credential where to send and run the create file methods.
 In future there may be the designation change so i think it is best way that to ask the user_id where i have to send.

Reply:
In second step i am using foreign key only

Reply:
This makes sense, however instead of storing the username in the model itself, you can actually query the 'HoldsDesignation' model to check for the username of the designation holder that you wish to send your file to.

Reply:
Nothing is hardcode, you can check by any user but there must be entry of designation in database.Ex AG,PK,VKJ has entry and all students. If any suggestionfrom yourside  then i am ready to move with that.

Reply:
I am sure nothing is hardcoded indeed, and this should work as expected. I was suggesting that you can actually store the designation in the model that you wish to send your file to and for extracting the username you can use. 
`python
ExtraInfo.objects.filter(designation=designation_name).name
` 
or something similar to that. If you are going to stick with this, however, please add a few lines of comment in the code indicating the reason as you stated above.